### PR TITLE
optimize communicator merge sparse gradient test=develop

### PR DESCRIPTION
**Pull request by jacquesqiao**
增加一个开关 communicator_merge_sparse_grad，用来决定是否对稀疏梯度进行merge之后再发送。默认是打开的状态。
http://agroup.baidu.com/paddlepaddle/view/office/1826179

### CODE REVIEW

File: python/paddle/fluid/__init__.py [hunk 1]
```diff
@@ -175,6 +175,7 @@ def __bootstrap__():
         read_env_flags.append('communicator_max_merge_var_num')
         read_env_flags.append('communicator_fake_rpc')
         read_env_flags.append('communicator_send_wait_times')
+        read_env_flags.append('communicator_merge_sparse_grad')
```
Comment (guru4elephant) [inline python/paddle/fluid/__init__.py:178]: maybe you should add flag document for this flag. I'm ok with this flag, do you have any recommendation about how to set this value?

File: paddle/fluid/operators/distributed/communicator.cc [hunk 2]
```diff
@@ -40,6 +40,8 @@ DEFINE_int32(communicator_max_merge_var_num, 20,
              "max var num to merge and send");
 DEFINE_bool(communicator_fake_rpc, false,
             "fake mode does not really send any thing");
+DEFINE_bool(communicator_merge_sparse_grad, true,
+            "merge sparse gradient before send");
```
Comment (xsrobin) [inline paddle/fluid/operators/distributed/communicator.cc]: before sending

Comment (jacquesqiao) [reply paddle/fluid/operators/distributed/communicator.cc]: done

**Review by guru4elephant — APPROVED** [review]
LGTM

**Review by xsrobin — APPROVED** [review]

**Pull request merged**